torchreid/engine/engine.py

```diff
@@ -19,6 +20,33 @@
                              visualize_ranked_results)
 
 
+EpochIntervalToValue = namedtuple('EpochIntervalToValue', ['first', 'last', 'value_inside', 'value_outside'])
+
+def _get_cur_action_from_epoch_interval(epoch_interval, epoch):
+    assert isinstance(epoch_interval, EpochIntervalToValue)
+    if epoch_interval.first is None and epoch_interval.last is None:
+        raise RuntimeError(f'Wrong epoch_interval {epoch_interval}')
+
+    if epoch_interval.first is not None and epoch < epoch_interval.first:
+        print(f'_get_cur_action_from_epoch_interval: since epoch={epoch} < epoch_interval.first={epoch_interval.first}, '
+              f'return value_outside={epoch_interval.value_outside}')
+        return epoch_interval.value_outside
+    if epoch_interval.last is not None and epoch > epoch_interval.last:
+        print(f'_get_cur_action_from_epoch_interval: since epoch={epoch} > epoch_interval.last={epoch_interval.last}, '
+              f'return value_outside={epoch_interval.value_outside}')
+        return epoch_interval.value_outside
+
+    print(f'_get_cur_action_from_epoch_interval: return value_inside={epoch_interval.value_inside}')
+    return epoch_interval.value_inside
+
+def get_initial_lr_from_checkpoint(filename):
```

Comment:
I think it's an inappropriate place for this function. It's not related to Engine

Reply:
Kirill, it is really related to the method how `initial_lr` is stored in the checkpoint.
But this is defined in one place in the code -- in the function `save_model` in `Engine`
I think it is a really wrong idea to separate the method of writing `initial_lr` field to checkpoint and the method of loading it from checkpoint.
(Otherwise later when somebody change saving it could forget to change loading, and NNCF pipeline will be broken).
I could make it a static function of `Engine` next to the method `save_model` to make it more clear.